### PR TITLE
Adjust QueryRepresentation and SPARQL query of any / regardless state

### PR DESCRIPTION
**Pull request by guergana**
+ added enum type PropertyValueRelation
+ updated condition instances to use PropertyValueRelation
+ updated QueryObjectBuilder by adding "any value" case
+ added test to confirm generation of any value case query

### CODE REVIEW

File: src/sparql/QueryObjectBuilder.ts [hunk 1]
```diff
@@ -52,6 +54,46 @@ export default class QueryObjectBuilder {
 			},
 		];
 
+		if ( queryRepresentation.condition.propertyValueRelation === PropertyValueRelation.NotMatching ) {
```
Comment (Ladsgroup) [inline src/sparql/QueryObjectBuilder.ts]: This is repeating a lot of query building logic from the above. Can't it have some cleaner way to do it?

Comment (guergana) [reply src/sparql/QueryObjectBuilder.ts]: Hi, I am aware of this, will do in next iteration.

File: tests/unit/sparql/buildQuery.spec.ts [hunk 2]
```diff
@@ -1,14 +1,51 @@
...
+		} } ) ).toEqual( expectedQuery );
+	} );
+
+	it( 'builds a query from a property with "any value" selected', () => {
+		const propertyId = 'P666';
+		const propertyValueRelation = PropertyValueRelation.Regardless;
+
+		expect( buildQuery( { condition: {
+			propertyId,
+			value: '',
```
Comment (guergana) [inline tests/unit/sparql/buildQuery.spec.ts:23]: I first tried not sending the value but it got messy with the types from the Terms tripleObject in `src/sparql/QueryObjectBuilder.ts `. Happy to read suggestions.

Comment (micgro42) [reply tests/unit/sparql/buildQuery.spec.ts:23]: Mh, I think it is okay to use it like that for now. Maybe we could also use `null`, but an empty string is also just fine here as that should not be possible by user input.

File: src/data-model/PropertyValueRelation.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+enum PropertyValueRelation {
```
Comment (micgro42) [inline src/data-model/PropertyValueRelation.ts:1]: I like it!

**Review by micgro42 — APPROVED** [review]

**Pull request merged**